pkg/providers/snow/reconciler/validatorbuilder.go

```diff
@@ -0,0 +1,30 @@
+package reconciler
+
+import (
+	"context"
+
+	"github.com/aws/eks-anywhere/pkg/aws"
+	"github.com/aws/eks-anywhere/pkg/providers/snow"
+)
+
+type ClientRegistry interface {
+	Get(ctx context.Context) (aws.Clients, error)
+}
+
+type validatorBuilder struct {
+	clientRegistry ClientRegistry
+}
+
+func NewValidatorBuilder(clientRegistry ClientRegistry) *validatorBuilder {
+	return &validatorBuilder{
+		clientRegistry: clientRegistry,
+	}
+}
+
+func (b *validatorBuilder) Build(ctx context.Context) (snow.Validator, error) {
+	deviceClientMap, err := b.clientRegistry.Get(ctx)
```

Comment:
I wonder if we should call this from inside the "validate" methods in the validator
I think that would allow us to avoid having a validator builder, right? We could just inject the validator directly in the reconciler